syncer/syncer.go

```diff
@@ -209,7 +212,7 @@ func NewSyncer(cfg *config.SubTaskConfig, etcdClient *clientv3.Client) *Syncer {
 	syncer.sqlOperatorHolder = operator.NewHolder()
 	syncer.readerHub = streamer.GetReaderHub()
 
-	if cfg.IsSharding {
+	if cfg.ShardMode == config.ShardPessimistic {
```

Comment:
Will `cfg.ShardMode == config.ShardPessimistic` but `cfg.IsSharding==false` need `syncer.sgk`? Also not so important, we can omit this in this PR.

Reply:
I want to give `shard-mode` higher priority and may to mark `is-sharding` as _deprecated_.